landing/script/signup.js

```diff
@@ -1,91 +1,126 @@
-  pwdCorrect: document.querySelector("input[name=pwdCorrect]"),
-};
-const { email, password, pwdCorrect } = userInput;
+//DOM, 유사 배열들을 배열로 바꿈
+const $loginForm = document.querySelector(".sign-form");
+const $emailInput = document.querySelector("input[name=email]");
+const $passwordInput = document.querySelector("input[name=password]");
+const $pwdCorrectInput = document.querySelector("input[name=pwdCorrect]");
+const $formInputList = [...document.querySelectorAll(".sign-input")];
+const $errorMsgList = [...document.querySelectorAll(".error-msg-box")];
 
-// const email = document.querySelector("input[name=email]");
-// const password = document.querySelector("input[name=password]");
 // 이메일과 비밀번호를 검사하는 정규 표현식 선언
 const regexEmail = /^[A-Za-z0-9_\.\-]+@[A-Za-z0-9\-]+\.[A-za-z0-9\-]+/;
 const regexPwd = /^(?=.*[A-Za-z])(?=.*\d)[A-Za-z\d]{8,}$/;
-//변수 button에 로그인 버튼 선언
-const loginForm = document.querySelector(".sign-form");
-//변수 errorMsg 선언
-const errorMsgEmail = document.querySelector("#error-msg-email");
-const errorMsgPwd = document.querySelector("#error-msg-pwd");
-const errorMsgPwdCorrect = document.querySelector("#error-msg-pwd-correct");
+
+//key값 선언
+const keyEmail = "test@codeit.com";
+const keyPassword = "codeit101";
+
 //변수 eyeButtom과 eyeImg 선언
 const eyeButton = document.querySelector(".eye-button");
 const eyeButtonCorrect = document.querySelector(".eye-button-correct");
 const eyeImg = document.querySelector(".eye-button img");
 const eyeImgCorrect = document.querySelector(".eye-button-correct img");
 
-//이메일 포커스 아웃 이벤트 발생 시 유효성 검사
-email.addEventListener("focusout", (event) => {
-  if (!event.target.value) {
-    email.style.borderColor = "black";
-    errorMsgEmail.style.color = "red";
-    errorMsgEmail.innerHTML = "이메일을 입력해주세요.";
-  } else if (!regexEmail.test(event.target.value)) {
-    email.style.borderColor = "red";
-    errorMsgEmail.style.color = "red";
-    errorMsgEmail.innerHTML = "올바른 이메일 주소가 아닙니다.";
-  } else {
-    errorMsgEmail.innerHTML = "";
-  }
-});
-//비밀번호 포커스 아웃 인벤트 발생 시 유효성 검사
-password.addEventListener("focusout", (event) => {
-  if (!event.target.value) {
-    password.style.borderColor = "black";
-    errorMsgPwd.style.color = "red";
-    errorMsgPwd.innerHTML = "비밀번호를 입력해주세요.";
-  } else if (!regexPwd.test(event.target.value)) {
-    password.style.borderColor = "red";
-    errorMsgPwd.style.color = "red";
-    errorMsgPwd.innerHTML = "올바른 비밀번호가 아닙니다.";
-  } else {
-    errorMsgPwd.innerHTML = "";
+//유효성 검사 함수
+const validator = {
+  required: {
+    validate: (value) => Boolean(value),
+    message: (name) => `${name} 입력해주세요`,
+  },
+  needEmailChar: {
+    validate: (value) => regexEmail.test(value),
+    message: `올바른 이메일 주소를 입력해주세요`,
+  },
+  needPasswordChar: {
+    validate: (value) => regexPwd.test(value),
+    message: `비밀번호는 문자와 숫자가 들어가고 8자리 이상이어야 합니다`,
+  },
+  needPwdCorrect: {
+    validate: (value) => $passwordInput.value === value,
+    message: `비밀번호와 다릅니다`,
+  },
+};
```

Comment:
아마 이부분이 가장 큰 어려움이 있지 않을까 생각했어요

외부에서 값을 사용하는것보다 이렇게 사용해보는건 어때요?
```suggestion
  needPwdCorrect: {
    validate: ([compare, target]) => compare === target,
    message: `비밀번호와 다릅니다`,
  },
};
```